learn/advanced/filtering_and_faceted_search.md

```diff
@@ -45,29 +33,37 @@ Suppose you have a collection of movies containing the following fields:
 ]
 ```
 
-If you want to filter results based on the `director` and `genres` attributes, you must add them to the [`filterableAttributes` list](/reference/api/settings.md#filterable-attributes):
+If you want to filter results based on the `director` and `genres` attributes, you must add them to the `filterableAttributes` list:
 
 <CodeSamples id="faceted_search_update_settings_1" />
 
-## Using filters
+**This step is mandatory and cannot be done at search time**  because Meilisearch needs to prepare an index before you can use filters. Updating `filterableAttributes` may take a significant amount of time depending on your dataset size.
+
+::: note
+By default, `filterableAttributes` is empty. Filters do not work without first explicitly adding attributes to the `filterableAttributes` list.
+:::
+
+Filters work with numeric and string values. Empty fields or fields containing an empty array will be ignored.
 
-Once you have configured `filterableAttributes`, you can start using [the `filter` search parameter](/reference/api/search.md#filter). Search parameters are added to at search time, that is, when a user searches your dataset.
+Filters do not work with [`NaN`](https://en.wikipedia.org/wiki/NaN) and infinite values such as `inf` and `-inf` as they are [not supported by JSON](https://en.wikipedia.org/wiki/JSON#Data_types). It is possible to filter infinite and `NaN` values if you parse them as strings, except when handling [`_geo` fields](/learn/advanced/geosearch.md#preparing-documents-for-location-based-search).
+
+## Filter basics
```

Comment:
What if we called this `The filter search parameter` or something like that?

I think that the article has a slight problem with heading titles. This is called "Filter basics" but there is no "Advanced use of filters" or equivalent. The following heading is called "Using filters", but this section seems to contain most if not all of the information necessary to use filters. If possible, I would like you to rethink the names of both of these sections, considering what value they will bring to the user. Why would a user want to jump to this section or the other? What would they be looking for?